src/Core/Feature.js

```diff
@@ -0,0 +1,154 @@
+import Extent from 'Core/Geographic/Extent';
+
+function defaultExtent(crs) {
+    return new Extent(crs, Infinity, -Infinity, Infinity, -Infinity);
+}
+
+function unionExtent(extent1, extent2) {
+    if (extent1.west == Infinity) {
+        extent1.copy(extent2);
+    } else {
+        extent1.union(extent2);
+    }
+}
+
+function _extendBuffer(feature, size) {
+    feature.vertices.length += size * feature.size;
+    if (feature.normals) {
+        feature.normals.length = feature.vertices.length;
+    }
+}
+
+class FeatureGeometry {
+    constructor(feature) {
+        this.extent = feature.extent ? defaultExtent(feature.crs) : undefined;
+        this.indices = [];
+        this._feature = feature;
+        this.properties = {};
+    }
+
+    startSubGeometry(count) {
+        const last = this.indices.length - 1;
+        const offset = last > -1 ?
+            this.indices[last].offset + this.indices[last].count :
+            this._feature.vertices.length / this._feature.size;
+        this.indices.push({ offset, count });
+        _extendBuffer(this._feature, count);
+    }
+
+    closeSubGeometry(count) {
+        const last = this.indices.length - 1;
+        const offset = last > -1 ?
+            this.indices[last].offset + this.indices[last].count :
+            this._feature.vertices.length / this._feature.size - count;
+        this.indices.push({ offset, count });
+    }
+
+    pushCoordinates(coord) {
+        if (coord.crs !== this._feature.crs) {
+            coord.as(this._feature.crs, coord);
+        }
+        if (this._feature.normals) {
+            coord.geodesicNormal.toArray(this._feature.normals, this._feature._pos);
+        }
+
+        this._feature._pushValues(this._feature, coord._values[0], coord._values[1], coord._values[2]);
+        // expand extent if present
+        if (this.extent) {
+            this.extent.expandByCoordinates(coord);
+        }
+    }
+
+    pushCoordinatesValues(value0, value1, value2) {
```

Comment:
So I get that this is the method using `Extent#expandByValuesCoordinates`, but now how do you see the usage of this ? Can't we only allow the push of complete coordinates ?

Reply:
This function will be used when the projection is identical and there's no conversion. This avoids having a useless `Coordinates` object in the parser. This will be used in `VectorTileParser`.